manifests/renew.pp

```diff
@@ -86,13 +90,52 @@
   ]).filter | $arg | { $arg =~ NotUndef and $arg != [] }
   $command = join($_command, ' ')
 
-  cron { 'letsencrypt-renew':
-    ensure   => $cron_ensure,
-    command  => $command,
-    user     => 'root',
-    hour     => $cron_hour,
-    minute   => $cron_minute,
-    monthday => $cron_monthday,
+  case $timer_type {
+    'cron': {
+      cron { 'letsencrypt-renew':
+        ensure   => $cron_ensure,
+        command  => $command,
+        user     => 'root',
+        hour     => $cron_hour,
+        minute   => $cron_minute,
+        monthday => $cron_monthday,
+      }
+    }
+    'systemd': {
+      if $facts['systemd'] {
+        $systemd_ensure = present
+      } else {
+        $systemd_ensure = absent
+      }
+      systemd::unit_file { 'letsencrypt.timer':
+        ensure  => $systemd_ensure,
+        content => template('letsencrypt/letsencrypt.timer.erb'),
```

Comment:
please use epp instead of erb for new templates.

Reply:
I have done this. I do not know how to check for an empty array in epp. I have it working, but not sure it is correct.